### ISSUE TITLE
[Test Failure]RecoveryIT IndexStatsIT Tests Failure

### ISSUE THREAD

**Issue by Bukhtawar** [opensearch-project/OpenSearch#5766]
**Describe the bug**
[org.opensearch.upgrades.RecoveryIT.testRelocationWithConcurrentIndexing](https://build.ci.opensearch.org/job/gradle-check/9089/testReport/junit/org.opensearch.upgrades/RecoveryIT/testRelocationWithConcurrentIndexing/)
[org.opensearch.upgrades.RecoveryIT.testRelocationWithConcurrentIndexing](https://build.ci.opensearch.org/job/gradle-check/9089/testReport/junit/org.opensearch.upgrades/RecoveryIT/testRelocationWithConcurrentIndexing_2/)
[org.opensearch.upgrades.RecoveryIT.testRelocationWithConcurrentIndexing](https://build.ci.opensearch.org/job/gradle-check/9089/testReport/junit/org.opensearch.upgrades/RecoveryIT/testRelocationWithConcurrentIndexing_3/)
[org.opensearch.upgrades.RecoveryIT.testRelocationWithConcurrentIndexing](https://build.ci.opensearch.org/job/gradle-check/9089/testReport/junit/org.opensearch.upgrades/RecoveryIT/testRelocationWithConcurrentIndexing_4/)
[org.opensearch.indices.replication.SegmentReplicationIT.testStartReplicaAfterPrimaryIndexesDocs](https://build.ci.opensearch.org/job/gradle-check/9089/testReport/junit/org.opensearch.indices.replication/SegmentReplicationIT/testStartReplicaAfterPrimaryIndexesDocs/)
[org.opensearch.indices.replication.SegmentReplicationRelocationIT.testIndexReopenClose](https://build.ci.opensearch.org/job/gradle-check/9089/testReport/junit/org.opensearch.indices.replication/SegmentReplicationRelocationIT/testIndexReopenClose/)
[org.opensearch.indices.stats.IndexStatsIT.testFilterCacheStats](https://build.ci.opensearch.org/job/gradle-check/9089/testReport/junit/org.opensearch.indices.stats/IndexStatsIT/testFilterCacheStats/)

**Comment by Bukhtawar** [comment]
Test Result (19 failures / +15)org.opensearch.backwards.IndexingIT.testUpdateSnapshotStatusorg.opensearch.backwards.IndexingIT.testSeqNoCheckpointsorg.opensearch.backwards.IndexingIT.testIndexVersionPropagationorg.opensearch.backwards.IndexingIT.classMethodorg.opensearch.backwards.IndexingIT.testUpdateSnapshotStatusorg.opensearch.backwards.IndexingIT.testSeqNoCheckpointsorg.opensearch.backwards.IndexingIT.testIndexVersionPropagationorg.opensearch.backwards.IndexingIT.classMethodorg.opensearch.backwards.IndexingIT.testUpdateSnapshotStatusorg.opensearch.backwards.IndexingIT.testSeqNoCheckpointsorg.opensearch.backwards.IndexingIT.testIndexVersionPropagationorg.opensearch.backwards.IndexingIT.classMethodorg.opensearch.backwards.IndexingIT.testSeqNoCheckpointsorg.opensearch.backwards.IndexingIT.testIndexVersionPropagationorg.opensearch.upgrades.RecoveryIT.testRelocationWithConcurrentIndexingorg.opensearch.upgrades.RecoveryIT.testRelocationWithConcurrentIndexingorg.opensearch.upgrades.RecoveryIT.testRelocationWithConcurrentIndexingorg.opensearch.upgrades.RecoveryIT.testRelocationWithConcurrentIndexing
--


[Test Result](https://build.ci.opensearch.org/job/gradle-check/9085/testReport/) (19 failures / +15)
[org.opensearch.backwards.IndexingIT.testUpdateSnapshotStatus](https://build.ci.opensearch.org/job/gradle-check/9085/testReport/junit/org.opensearch.backwards/IndexingIT/testUpdateSnapshotStatus/)
[org.opensearch.backwards.IndexingIT.testSeqNoCheckpoints](https://build.ci.opensearch.org/job/gradle-check/9085/testReport/junit/org.opensearch.backwards/IndexingIT/testSeqNoCheckpoints/)
[org.opensearch.backwards.IndexingIT.testIndexVersionPropagation](https://build.ci.opensearch.org/job/gradle-check/9085/testReport/junit/org.opensearch.backwards/IndexingIT/testIndexVersionPropagation/)
[org.opensearch.backwards.IndexingIT.classMethod](https://build.ci.opensearch.org/job/gradle-check/9085/testReport/junit/org.opensearch.backwards/IndexingIT/classMethod/)
[org.opensearch.backwards.IndexingIT.testUpdateSnapshotStatus](https://build.ci.opensearch.org/job/gradle-check/9085/testReport/junit/org.opensearch.backwards/IndexingIT/testUpdateSnapshotStatus_2/)
[org.opensearch.backwards.IndexingIT.testSeqNoCheckpoints](https://build.ci.opensearch.org/job/gradle-check/9085/testReport/junit/org.opensearch.backwards/IndexingIT/testSeqNoCheckpoints_2/)
[org.opensearch.backwards.IndexingIT.testIndexVersionPropagation](https://build.ci.opensearch.org/job/gradle-check/9085/testReport/junit/org.opensearch.backwards/IndexingIT/testIndexVersionPropagation_2/)
[org.opensearch.backwards.IndexingIT.classMethod](https://build.ci.opensearch.org/job/gradle-check/9085/testReport/junit/org.opensearch.backwards/IndexingIT/classMethod_2/)
[org.opensearch.backwards.IndexingIT.testUpdateSnapshotStatus](https://build.ci.opensearch.org/job/gradle-check/9085/testReport/junit/org.opensearch.backwards/IndexingIT/testUpdateSnapshotStatus_3/)
[org.opensearch.backwards.IndexingIT.testSeqNoCheckpoints](https://build.ci.opensearch.org/job/gradle-check/9085/testReport/junit/org.opensearch.backwards/IndexingIT/testSeqNoCheckpoints_3/)
[org.opensearch.backwards.IndexingIT.testIndexVersionPropagation](https://build.ci.opensearch.org/job/gradle-check/9085/testReport/junit/org.opensearch.backwards/IndexingIT/testIndexVersionPropagation_3/)
[org.opensearch.backwards.IndexingIT.classMethod](https://build.ci.opensearch.org/job/gradle-check/9085/testReport/junit/org.opensearch.backwards/IndexingIT/classMethod_3/)
[org.opensearch.backwards.IndexingIT.testSeqNoCheckpoints](https://build.ci.opensearch.org/job/gradle-check/9085/testReport/junit/org.opensearch.backwards/IndexingIT/testSeqNoCheckpoints_4/)
[org.opensearch.backwards.IndexingIT.testIndexVersionPropagation](https://build.ci.opensearch.org/job/gradle-check/9085/testReport/junit/org.opensearch.backwards/IndexingIT/testIndexVersionPropagation_4/)
[org.opensearch.upgrades.RecoveryIT.testRelocationWithConcurrentIndexing](https://build.ci.opensearch.org/job/gradle-check/9085/testReport/junit/org.opensearch.upgrades/RecoveryIT/testRelocationWithConcurrentIndexing/)
[org.opensearch.upgrades.RecoveryIT.testRelocationWithConcurrentIndexing](https://build.ci.opensearch.org/job/gradle-check/9085/testReport/junit/org.opensearch.upgrades/RecoveryIT/testRelocationWithConcurrentIndexing_2/)
[org.opensearch.upgrades.RecoveryIT.testRelocationWithConcurrentIndexing](https://build.ci.opensearch.org/job/gradle-check/9085/testReport/junit/org.opensearch.upgrades/RecoveryIT/testRelocationWithConcurrentIndexing_3/)
[org.opensearch.upgrades.RecoveryIT.testRelocationWithConcurrentIndexing](https://build.ci.opensearch.org/job/gradle-check/9085/testReport/junit/org.opensearch.upgrades/RecoveryIT/testRelocationWithConcurrentIndexing_4/)

**Comment by mch2** [comment]
Moving 2.5 specific tests to a separate issue.

**Comment by kotwanikunal** [comment]
Segment Replication tests have been muted and are being root caused separately. 
We are seeing them post the feature flag merge: #5211

**Comment by dreamer-89** [comment]
I think the BWC tests are broken with this commit https://github.com/opensearch-project/OpenSearch/issues/5282 which introduced `replicated` boolean [flag](https://github.com/opensearch-project/OpenSearch/pull/5282/files#diff-d449776578711bd20edfeb168cc019d6e049614ef77bd190611253637ed5b79cR715) in `ReplicationTracker`. 
```
»  Caused by: org.opensearch.transport.RemoteTransportException: [v2.6.0-1][127.0.0.1:51603][internal:index/shard/recovery/handoff_primary_context]
»  Caused by: java.lang.IllegalStateException: unexpected byte [0x6c]
»       at org.opensearch.common.io.stream.StreamInput.readBoolean(StreamInput.java:588) ~[opensearch-2.6.0-SNAPSHOT.jar:2.6.0-SNAPSHOT]
»       at org.opensearch.common.io.stream.StreamInput.readBoolean(StreamInput.java:578) ~[opensearch-2.6.0-SNAPSHOT.jar:2.6.0-SNAPSHOT]
```

**Comment by anasalkouz** [comment]
Should be resolved. Closing the issue.

**Comment by andrross** [comment]
@saratvemulapalli I'm going to resolve this as I believe there was a version check mismatch fixed in #7672 that caused the failure you linked. Please reopen if that's wrong.

**Comment by saratvemulapalli** [comment]
Thanks @andrross. Rebase fixed it for me.
